tests/007-datagrams.cpp

```diff
@@ -160,87 +151,115 @@ namespace oxen::quic::test
                         return 0;
                     };
 
-            opt::enable_datagrams greedy{true, Splitting::GREEDY};
+            dgram_data_callback recv_dgram_cb = [&](bstring_view) {
+                log::debug(log_cat, "Calling endpoint receive datagram callback... data received...");
+
+                data_promise.set_value(true);
+            };
+
+            opt::enable_datagrams default_gram{};
+
             opt::local_addr server_local{};
             opt::local_addr client_local{};
 
             auto server_tls = GNUTLSCreds::make("./serverkey.pem"s, "./servercert.pem"s, "./clientcert.pem"s);
             auto client_tls = GNUTLSCreds::make("./clientkey.pem"s, "./clientcert.pem"s, "./servercert.pem"s);
             client_tls->set_client_tls_policy(outbound_tls_cb);
 
-            auto server_endpoint = test_net.endpoint(server_local, greedy);
+            auto server_endpoint = test_net.endpoint(server_local, default_gram, recv_dgram_cb);
             REQUIRE_NOTHROW(server_endpoint->listen(server_tls));
 
             opt::remote_addr client_remote{"127.0.0.1"s, server_endpoint->local().port()};
 
-            auto client_endpoint = test_net.endpoint(client_local, greedy);
-            auto conn_interface = client_endpoint->connect(client_remote, client_tls);
+            auto client = test_net.endpoint(client_local, default_gram);
+            auto conn_interface = client->connect(client_remote, client_tls);
 
             REQUIRE(tls_future.get());
-            REQUIRE(conn_interface->datagrams_enabled());
-            REQUIRE(conn_interface->packet_splitting_enabled());
-            REQUIRE(conn_interface->packet_splitting_policy() == Splitting::GREEDY);
 
-            std::this_thread::sleep_for(5ms);
+            REQUIRE(server_endpoint->datagrams_enabled());
+            REQUIRE(client->datagrams_enabled());
 
+            REQUIRE(conn_interface->datagrams_enabled());
+            REQUIRE_FALSE(conn_interface->packet_splitting_enabled());
             REQUIRE(conn_interface->get_max_datagram_size() < MAX_GREEDY_PMTUD_UDP_PAYLOAD);
 
+            conn_interface->send_datagram(msg);
+
+            std::this_thread::sleep_for(500ms);
+
+            REQUIRE(data_future.get());
+
             test_net.close();
         };
     };
 
-    TEST_CASE("007 - Datagram support: Execute No Splitting Policy", "[007][datagrams][execute]")
+    TEST_CASE("007 - Datagram support: Execute, Packet Splitting Enabled", "[007][datagrams][execute][split]")
     {
         SECTION("Simple datagram transmission")
         {
             Network test_net{};
-            auto msg = "hello from the other siiiii-iiiiide"_bsv;
+            std::string data{};
+            char v = 0;
 
-            std::promise<bool> tls_promise, data_promise;
-            std::future<bool> tls_future = tls_promise.get_future(), data_future = data_promise.get_future();
+            while (data.size() < 2000)
```

Comment:
This number may as well reflect max datagram size.

Reply:
good thought, will reflect in the updated 007 test when i add the rotating buffer test